salt/metalk8s/kubeadm/init/certs/ca.sls

```diff
@@ -0,0 +1,53 @@
+{%- from "metalk8s/map.jinja" import ca with context %}
+
+Create CA private key:
+  x509.private_key_managed:
+    - name: {{ ca.cert.signing_private_key }}
+    - bits: {{ ca.cert.bits }}
+    - user: root
+    - group: root
+    - mode: 600
+    - makedirs: True
+    - dir_mode: 755
+
+Generate CA certificate:
+  x509.certificate_managed:
+    {%- for key, value in ca.cert.items() %}
+    - {{ key }}: {{ value }}
+    {%- endfor %}
+    - user: root
+    - group: root
+    - mode: 644
+    - makedirs: True
+    - dir_mode: 755
+    - require:
+      - x509: Create CA private key
+
+# TODO: Find a better way to advetise CA server
+#       (maybe not advertise if we got already one)
```

Comment:
I think you can remove this TODO. If we execute this Salt state, we expect this host to advertise itself as the CA server, even if there is already another one.